qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/VisitTypeSteps.java

```diff
@@ -0,0 +1,104 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.RestClient;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.VisitTypeListPage;
+import org.openmrs.contrib.qaframework.page.VisitTypePage;
+
+import static org.junit.Assert.assertTrue;
+
+public class VisitTypeSteps extends Steps {
+
+    private AdministrationPage administrationPage;
+    private VisitTypePage visitTypePage;
+    private VisitTypeListPage visitTypeListPage;
+    private String visitTypeUuid;
+    private final String VISIT_TYPE_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String VISIT_TYPE_DESCRIPTION = "Description for Visit Type test workflow";
+
+    @Before(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void setup() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void teardown() {
+        quit();
+        RestClient.delete("visittype/" + visitTypeUuid, true);
+    }
+
+    @Given("User clicks the system administration link on the home page")
+    public void launchSystemAdministrationPage() {
+        administrationPage = homePage.goToAdministration();
+    }
+
+    @When("User clicks the manage Visit Types on the administration page")
+    public void launchAdvancedAdministrationPage() {
+        visitTypeListPage = administrationPage.goToVisitTypePage();
+    }
+
+    @And("User clicks the add visit type link on the manage visit type page")
+    public void clickOnAddVisitTypeLink() {
+        visitTypePage = visitTypeListPage.addVisitType();
+    }
+
+    @And("User fills the add visit type form")
+    public void fillAddVisitTypeForm(){
+        visitTypePage.setName(VISIT_TYPE_NAME);
+        visitTypePage.setDescription(VISIT_TYPE_DESCRIPTION);
+    }
+
+    @And ("User saves visit type")
+    public void saveVisitType() {
+        visitTypePage.save();
+    }
+
+    @Then("visit type is saved successfully")
+    public void systemAddsVisitType() {
+        assertTrue(textExists("Visit Type saved"));
+    }
+
+    @And("User clicks on already saved visit type")
+    public void clickAlreadySavedVisitType() {
+        visitTypePage = visitTypeListPage.clickAlreadySavedVisitType();
+    }
+
+    @And("User edits the visit type form")
+    public void editVisitType(){
+        visitTypePage.setName("new VisitTypeName");
```

Comment:
How is this method editting the function. Do you mind explaining here .

Reply:
It sets `new VisitTypeName` as a new  visitType name then after it saves it hence editing VisitType